src/boot/efi/boot.c

```diff
@@ -101,6 +101,53 @@ enum {
         TIMEOUT_TYPE_MAX    = UINT32_MAX,
 };
 
+#define _returns_nonnull_ __attribute__((__returns_nonnull__))
+#define _allocate_pool_ __attribute__((__malloc__, __malloc__(FreePool, 1), __warn_unused_result__))
+#define assert_alloc_return(p) ({ \
+        if (_likely_(p))          \
+                return p;         \
+        log_oom();                \
+        __builtin_trap();         \
+})
+
+_allocate_pool_ _returns_nonnull_ void *new_nofail(UINTN size) {
+        void *p = AllocatePool(size);
+        assert_alloc_return(p);
+}
+
+_allocate_pool_ _returns_nonnull_ void *new0_nofail(UINTN size) {
+        void *p = new_nofail(size);
+        ZeroMem(p, size);
+        return p;
+}
+
+_allocate_pool_ _returns_nonnull_ CHAR16 *pool_print_nofail(const CHAR16 *fmt, ...) {
+        assert(fmt);
+        va_list args;
+        va_start(args, fmt);
+        CHAR16 *p = VPoolPrint(fmt, args);
+        va_end(args);
+        assert_alloc_return(p);
+}
+
+_allocate_pool_ _returns_nonnull_ CHAR16 *strdup_nofail(const CHAR16 *src) {
+        assert(src);
+        CHAR16 *dup = StrDuplicate(src);
+        assert_alloc_return(dup);
+}
+
+_allocate_pool_ _returns_nonnull_ CHAR16 *stra_to_path_nofail(const CHAR8 *stra) {
+        assert(stra);
+        CHAR16 *path = stra_to_path(stra);
+        assert_alloc_return(path);
+}
+
+_allocate_pool_ _returns_nonnull_ CHAR16 *stra_to_str_nofail(const CHAR8 *stra) {
+        assert(stra);
+        CHAR16 *str = stra_to_str(stra);
```

Comment:
why define a wrapper for this one? the function is defined by us, let's just update it to use an aborting oom func